### ISSUE TITLE
[EPCO-5] Build a neural network model class

### ISSUE THREAD

**Issue by hatfield-c** [hatfield-c/epco#5]
There should be a single class from which N can be instantiated from.

The neural network of the class will be composed only of dense layers followed by ReLU activation layers
- Final dense layer is not followed by ReLU layer
- Final dense layer has one output
- There will likely be more needed methods related to search i.e. gradients

**Comment by hatfield-c** [comment]
The neural network has been modified for the time being to use a final smooth sigmoid layer, such that it can train more easily with the cross entropy loss function. This change should only effect the training process - after training, the pre-sigmoid network can be extracted if desired.

**Comment by hatfield-c** [comment]
This issue is now complete and will thus be closed.
